### PR TITLE
[Widgets] refactor call api-inference function

### PR DESCRIPTION
**Pull request by mishig25**
Refactors call api-inference function. See [files tab](https://github.com/huggingface/hub-docs/pull/1034/files)

### CODE REVIEW

**Review by julien-c — APPROVED** [review]
looks simpler / more compact indeed

File: js/src/lib/components/InferenceWidget/shared/helpers.ts [hunk 1]
```diff
@@ -52,16 +52,42 @@ export async function getBlobFromUrl(url: string): Promise<Blob> {
...
+	status: "error";
+}
+
+interface CacheNotFound {
+	status: "cache not found";
+}
+
+type Result<T> = Success<T> | LoadingModel | Error | CacheNotFound;
+
+export async function getResponse<T>(
```
Comment (SBrandeis) [inline js/src/lib/components/InferenceWidget/shared/helpers.ts:80]: Nitpicking, but `getResponse` is a very generic name

**Review by SBrandeis — APPROVED** [review]
Cleaner 👍

File: js/src/lib/components/InferenceWidget/shared/helpers.ts (hunk 1 above)
Comment (mishig25) [reply js/src/lib/components/InferenceWidget/shared/helpers.ts:80]: maybe `callInferenceApi` ?

Comment (mishig25) [reply js/src/lib/components/InferenceWidget/shared/helpers.ts]: handled in https://github.com/huggingface/hub-docs/pull/1034/commits/e16171ce99dfce6014a1557a17c77e98f168a203

**Pull request merged**